apps/prairielearn/src/components/Navbar.html.ts

```diff
@@ -0,0 +1,860 @@
+import { flash, type FlashMessageType } from '@prairielearn/flash';
+import { html, unsafeHtml } from '@prairielearn/html';
+
+import { config } from '../lib/config.js';
+
+import { IssueBadge } from './IssueBadge.html.js';
+import type { NavbarType, NavPage, NavSubPage } from './Navbar.types.js';
+import { ContextNavigation } from './NavbarContext.html.js';
+
+export function Navbar({
+  resLocals,
+  navPage,
+  navSubPage,
+  navbarType,
+}: {
+  resLocals: Record<string, any>;
+  navPage?: NavPage;
+  navSubPage?: NavSubPage;
+  navbarType?: NavbarType;
+}) {
+  const { __csrf_token, course, urlPrefix, homeUrl } = resLocals;
+  navPage ??= resLocals.navPage;
+  navSubPage ??= resLocals.navSubPage;
+  navbarType ??= resLocals.navbarType;
+
+  return html`
+    ${config.devMode && __csrf_token
+      ? // Unit tests often need access to the CSRF token even when the page contains
+        // no form - for example, to confirm that a POST with a prohibited
+        // action is denied. For convenience, we include the CSRF token here, on
+        // all pages. We do this only in devMode and only for the purpose of
+        // testing.
+        html`
+          <!-- DO NOT RELY ON OR USE THIS CSRF TOKEN FOR ANYTHING OTHER THAN UNIT TESTS! -->
+          <span id="test_csrf_token" hidden>${__csrf_token}</span>
+        `
+      : ''}
+    ${config.announcementHtml
+      ? html`
+          <div
+            class="alert alert-${config.announcementColor ?? 'primary'} mb-0 rounded-0 text-center"
+          >
+            ${unsafeHtml(config.announcementHtml)}
+          </div>
+        `
+      : ''}
+
+    <nav class="navbar navbar-dark bg-dark navbar-expand-md" aria-label="Global navigation">
+      <a class="navbar-brand" href="${homeUrl}">
+        <span class="navbar-brand-label">PrairieLearn</span>
+        <span class="navbar-brand-hover-label text-secondary">
+          Go home <i class="fa fa-angle-right" aria-hidden="true"></i>
+        </span>
+      </a>
+      <button
+        class="navbar-toggler"
+        type="button"
+        data-toggle="collapse"
+        data-target=".navbar-collapse"
+        aria-expanded="false"
+        aria-label="Toggle navigation"
+      >
+        <span class="navbar-toggler-icon"></span>
+      </button>
+      <div id="course-nav" class="collapse navbar-collapse">
+        <ul class="nav navbar-nav mr-auto" id="main-nav">
+          ${NavbarByType({ resLocals, navPage, navSubPage, navbarType })}
+        </ul>
+
+        ${config.devMode
+          ? html`
+              <a
+                id="navbar-load-from-disk"
+                class="btn btn-success btn-sm"
+                href="${urlPrefix}/loadFromDisk"
+              >
+                Load from disk
+              </a>
+            `
+          : ''}
+        ${UserDropdownMenu({ resLocals, navPage, navbarType })}
+      </div>
+    </nav>
+
+    ${navbarType === 'instructor' && course?.announcement_html && course.announcement_color
```

Comment:
IMO this is clearer:

```suggestion
    ${navbarType === 'instructor' && course && course.announcement_html && course.announcement_color
```